daemon/metrics.go

```diff
@@ -0,0 +1,84 @@
+package daemon
+
+import (
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/dockerversion"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/go-metrics"
+)
+
+var (
+	containerActions   metrics.LabeledTimer
+	imageActions       metrics.LabeledTimer
+	containerCounter   metrics.LabeledCounter
+	daemonActions      metrics.LabeledTimer
+	networkActions     metrics.LabeledTimer
+	engineVersion      metrics.LabeledGauge
+	healthCheckCounter metrics.Counter
+)
+
+func init() {
+	ns := metrics.NewNamespace("engine", "daemon", metrics.Labels{
+		"version": dockerversion.Version,
+		"commit":  dockerversion.GitCommit,
+	})
+	containerActions = ns.NewLabeledTimer("container_actions", "The number of seconds it takes to process each container action", "action")
+	daemonActions = ns.NewLabeledTimer("daemon_actions", "The number of seconds it takes to process each daemon action", "action")
+	networkActions = ns.NewLabeledTimer("network_actions", "The number of seconds it takes to process each network action", "action")
+	engineVersion = ns.NewLabeledGauge("version", "The version and commit information for the engine process", metrics.Unit("info"),
+		"architecture",
+		"cpus",
+		"graph_driver",
+		"kernel",
+		"memory",
+		"os",
+	)
+	containerCounter = ns.NewLabeledCounter("container", "The total number of containers processed", "action")
+	healthCheckCounter = ns.NewCounter("health_check", "The total number of health checks executed")
+	imageActions = ns.NewLabeledTimer("image_actions", "The number of seconds it takes to process each image action", "action")
+	metrics.Register(ns)
+}
+
+func (daemon *Daemon) startMetrics(c *container.Container) {
+	ns := metrics.NewNamespace("container", "stat", metrics.Labels{
+		"id":    c.ID,
+		"image": c.Config.Image,
+	})
+
+	var (
+		cpuTime          = ns.NewLabeledGauge("cpu_usage", "The number of nanoseconds a container uses in cpu time", metrics.Nanoseconds, "type")
```

Comment:
I'm not a huge fan of the units type. I don't think people will take time to make a considered choice in practice, leaving us with inconsistent counters throughout the project.

However, I do see your point.

In this case, I am really not sure, since this is actually a gauge (may need to be a counter). We don't really get a guaranteed conversion here, since it will still be incremented as a float.